### PR TITLE
[Bugfix] setup data

### PR DESCRIPTION
**Pull request by MatthewK3023**

### CODE REVIEW

File: docs/source/user/quickstart.rst [hunk 1]
```diff
@@ -27,7 +27,7 @@ in file ``examples/example.py``
 .. code-block:: python
 
     core.Context.create(
-        username='{usr}', password='{pwd}', host='{decantercoreserver}')
+        username='gp', password='gp-admin', host='http://localhost:3000')
```
Comment (NoctisHong) [inline docs/source/user/quickstart.rst:30]: use this on document: "username='{usr}', password='{pwd}', host='{host:port}')"

File: examples/auto_time_series_example.py [hunk 2]
```diff
@@ -35,6 +35,25 @@ def main():
...
+        data_source=train_data.accessor,
+        data_columns=[
+            {
+                'id': 'date',
+                'data_type': 'datetime'
+            }],
+        )
+
+    train_data = client.setup(setup_input)
+
```
Comment (NoctisHong) [inline examples/auto_time_series_example.py:56]: Just put setup example on genesis ~

File: src/decanter/core/client.py [hunk 3]
```diff
@@ -48,9 +48,7 @@ def __init__(self, username, password, host):
         
 
     @staticmethod
-    def setup(
-            train_data, data_source, data_columns, data_id=None, callback=None,
-            eda=None, preprocessing=None, version=None, name=None):
+    def setup(setup_input, name=None):
         """Setup data reference.
 
         Create a DataUpload Job and scheduled the execution in CORO_TASKS list.
```
Comment (NoctisHong) [inline src/decanter/core/client.py:54]: Need to update all document part to fit deata setup

File: src/decanter/core/core_api/setup_input.py [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import json
+
+from decanter.core.core_api import CoreBody
+from decanter.core.enums.algorithms import Algo
+from decanter.core.enums.evaluators import Evaluator
+from decanter.core.enums import check_is_enum
+
+
+class SetupInput:
```
Comment (NoctisHong) [inline src/decanter/core/core_api/setup_input.py:9]: Need document comment

File: src/decanter/core/jobs/data_setup.py [hunk 5]
```diff
@@ -42,18 +41,17 @@ class DataSetup(DataUpload, Job):
...
     """
-    def __init__(self, train_data=None, setup_params=None, name=None):
+    def __init__(self, setup_input=None, name=None):
         """DataSetup Init.
 
         Args:
             train_data (:obj:`~decanter.core.jobs.data_setup.DataSetup)
             setup_params (:obj:`dict`, optional): Settings for setup data.
             name (:obj:`str`, optional): Name to track Job progress
         """
```
Comment (NoctisHong) [inline src/decanter/core/jobs/data_setup.py:51]: Need to update all document to fit setup data

File: src/decanter/core/jobs/task.py [hunk 6]
```diff
@@ -300,14 +300,15 @@ class SetupTask(CoreTask):
...
-    def __init__(self, params, name='Setup'):
+    def __init__(self, setup_input, name='Setup'):
         super().__init__(name=name)
-        self.params = params
+        self.setup_input = setup_input
 
     def run(self):
         """
         Execute setup data by sending the setup api.
         """
```
Comment (NoctisHong) [inline src/decanter/core/jobs/task.py:310]: Need to update document to fit new change

Comment (MatthewK3023) [reply src/decanter/core/jobs/task.py:310]: done

File: src/decanter/core/jobs/data_setup.py (hunk 5 above)
Comment (MatthewK3023) [reply src/decanter/core/jobs/data_setup.py:51]: done

File: src/decanter/core/core_api/setup_input.py (hunk 4 above)
Comment (MatthewK3023) [reply src/decanter/core/core_api/setup_input.py:9]: done

File: src/decanter/core/client.py (hunk 3 above)
Comment (MatthewK3023) [reply src/decanter/core/client.py:54]: done

File: examples/auto_time_series_example.py (hunk 2 above)
Comment (MatthewK3023) [reply examples/auto_time_series_example.py:56]: done

File: docs/source/user/quickstart.rst (hunk 1 above)
Comment (MatthewK3023) [reply docs/source/user/quickstart.rst:30]: done

File: src/decanter/core/core_api/setup_input.py [hunk 7]
```diff
@@ -7,6 +7,28 @@
...
+        data (:class:`~decanter.core.jobs.data_upload.DataUpload`):
+            Train data uploaded on Decanter Core server
+        setup_body (:class:`~decanter.core.core_api.body_obj.SetupBody`):
+            Request body for sending setup api.
+
+    Example:
+        .. code-block:: python
+
+            setup_input = SetupInput(
+                data = train_data,
```
Comment (NoctisHong) [inline src/decanter/core/core_api/setup_input.py:24]: upload_data

File: src/decanter/core/core_api/setup_input.py [hunk 8]
```diff
@@ -7,6 +7,28 @@
...
+                data_source=train_data.accessor,
```
Comment (NoctisHong) [inline src/decanter/core/core_api/setup_input.py:25]: ditto

File: src/decanter/core/jobs/task.py [hunk 9]
```diff
@@ -297,7 +297,7 @@ class SetupTask(CoreTask):
     V2 eda use, setting v2 eda result for the data, preparing for custom eda.
 
     Attributes:
-        params (dict):
+        params (:class:`~decanter.core.core_api.setup_input.SetupInput`)::
```
Comment (NoctisHong) [inline src/decanter/core/jobs/task.py:300]: setup_input

File: src/decanter/core/core_api/setup_input.py (hunk 7 above)
Comment (MatthewK3023) [reply src/decanter/core/core_api/setup_input.py]: done

File: src/decanter/core/core_api/setup_input.py (hunk 8 above)
Comment (MatthewK3023) [reply src/decanter/core/core_api/setup_input.py]: done

File: src/decanter/core/jobs/task.py (hunk 9 above)
Comment (MatthewK3023) [reply src/decanter/core/jobs/task.py]: done

**Review by NoctisHong — APPROVED** [review]

**Pull request merged**